[2023-10-29T20:20:15.305+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: random_names.streaming_to_kafka scheduled__2023-10-29T20:15:00+00:00 [queued]>
[2023-10-29T20:20:15.314+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: random_names.streaming_to_kafka scheduled__2023-10-29T20:15:00+00:00 [queued]>
[2023-10-29T20:20:15.315+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 1
[2023-10-29T20:20:15.376+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): streaming_to_kafka> on 2023-10-29 20:15:00+00:00
[2023-10-29T20:20:15.389+0000] {standard_task_runner.py:57} INFO - Started process 286 to run task
[2023-10-29T20:20:15.398+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'random_names', 'streaming_to_kafka', 'scheduled__2023-10-29T20:15:00+00:00', '--job-id', '39', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmp5l42srvs']
[2023-10-29T20:20:15.405+0000] {standard_task_runner.py:85} INFO - Job 39: Subtask streaming_to_kafka
[2023-10-29T20:20:15.575+0000] {task_command.py:415} INFO - Running <TaskInstance: random_names.streaming_to_kafka scheduled__2023-10-29T20:15:00+00:00 [running]> on host 895885866eac
[2023-10-29T20:20:15.935+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='random_names' AIRFLOW_CTX_TASK_ID='streaming_to_kafka' AIRFLOW_CTX_EXECUTION_DATE='2023-10-29T20:15:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-29T20:15:00+00:00'
[2023-10-29T20:20:16.515+0000] {logging_mixin.py:151} INFO - {'full name': 'Mr Steinar Hordvik', 'gender': 'male', 'location': 'Smedstusvingen 6557', 'city': 'Sponvika', 'country': 'Norway', 'postcode': '4885', 'latitude': '44.9332', 'longtitude': '-47.0289', 'email': 'steinar.hordvik@example.com'}
[2023-10-29T20:20:16.521+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]>: connecting to localhost:29092 [('127.0.0.1', 29092) IPv4]
[2023-10-29T20:20:16.521+0000] {conn.py:1205} INFO - Probing node bootstrap-0 broker version
[2023-10-29T20:20:16.522+0000] {conn.py:418} ERROR - Connect attempt to <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]> returned error 111. Disconnecting.
[2023-10-29T20:20:16.522+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]>: Closing connection. KafkaConnectionError: 111 ECONNREFUSED
[2023-10-29T20:20:16.570+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]>: connecting to localhost:29092 [('127.0.0.1', 29092) IPv4]
[2023-10-29T20:20:16.571+0000] {conn.py:418} ERROR - Connect attempt to <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]> returned error 111. Disconnecting.
[2023-10-29T20:20:16.571+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]>: Closing connection. KafkaConnectionError: 111 ECONNREFUSED
[2023-10-29T20:20:16.620+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv6 ('::1', 29092, 0, 0)]>: connecting to localhost:29092 [('::1', 29092, 0, 0) IPv6]
[2023-10-29T20:20:16.621+0000] {conn.py:418} ERROR - Connect attempt to <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv6 ('::1', 29092, 0, 0)]> returned error 99. Disconnecting.
[2023-10-29T20:20:16.621+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv6 ('::1', 29092, 0, 0)]>: Closing connection. KafkaConnectionError: 99 EADDRNOTAVAIL
[2023-10-29T20:20:16.621+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/scripts/script.py", line 50, in start_streaming
    producer = get_kafka_producer()
  File "/opt/airflow/dags/scripts/script.py", line 31, in get_kafka_producer
    producer = KafkaProducer(
  File "/home/airflow/.local/lib/python3.8/site-packages/kafka/producer/kafka.py", line 381, in __init__
    client = KafkaClient(metrics=self._metrics, metric_group_prefix='producer',
  File "/home/airflow/.local/lib/python3.8/site-packages/kafka/client_async.py", line 244, in __init__
    self.config['api_version'] = self.check_version(timeout=check_timeout)
  File "/home/airflow/.local/lib/python3.8/site-packages/kafka/client_async.py", line 900, in check_version
    raise Errors.NoBrokersAvailable()
kafka.errors.NoBrokersAvailable: NoBrokersAvailable
[2023-10-29T20:20:16.772+0000] {taskinstance.py:1398} INFO - Marking task as FAILED. dag_id=random_names, task_id=streaming_to_kafka, execution_date=20231029T201500, start_date=20231029T202015, end_date=20231029T202016
[2023-10-29T20:20:16.845+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 39 for task streaming_to_kafka (NoBrokersAvailable; 286)
[2023-10-29T20:20:16.854+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-10-29T20:20:16.868+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
